cmd/warm-backend-azure.go

```diff
@@ -49,150 +59,107 @@ func (az *warmBackendAzure) getDest(object string) string {
 	return destObj
 }
 
-func (az *warmBackendAzure) tier() azblob.AccessTierType {
-	for _, t := range azblob.PossibleAccessTierTypeValues() {
-		if strings.EqualFold(az.StorageClass, string(t)) {
-			return t
-		}
-	}
-	return azblob.AccessTierType("")
-}
-
-// FIXME: add support for remote version ID in Azure remote tier and remove
-// this. Currently it's a no-op.
-
 func (az *warmBackendAzure) Put(ctx context.Context, object string, r io.Reader, length int64) (remoteVersionID, error) {
-	blobURL := az.serviceURL.NewContainerURL(az.Bucket).NewBlockBlobURL(az.getDest(object))
-	// set tier if specified -
-	if az.StorageClass != "" {
-		if _, err := blobURL.SetTier(ctx, az.tier(), azblob.LeaseAccessConditions{}, azblob.RehydratePriorityStandard); err != nil {
-			return "", azureToObjectError(err, az.Bucket, object)
-		}
-	}
-	res, err := azblob.UploadStreamToBlockBlob(ctx, r, blobURL, azblob.UploadStreamToBlockBlobOptions{})
+	resp, err := az.clnt.UploadStream(ctx, az.Bucket, az.getDest(object), io.LimitReader(r, length), &azblob.UploadStreamOptions{
+		Concurrency: 4,
+		AccessTier:  az.tier(), // set tier if specified
+	})
 	if err != nil {
-		return "", azureToObjectError(err, az.Bucket, object)
+		return "", azureToObjectError(err, az.Bucket, az.getDest(object))
 	}
-	return remoteVersionID(res.Version()), nil
+	return remoteVersionID(*resp.VersionID), nil
 }
 
 func (az *warmBackendAzure) Get(ctx context.Context, object string, rv remoteVersionID, opts WarmBackendGetOpts) (r io.ReadCloser, err error) {
 	if opts.startOffset < 0 {
 		return nil, InvalidRange{}
 	}
-	blobURL := az.serviceURL.NewContainerURL(az.Bucket).NewBlobURL(az.getDest(object))
-	blob, err := blobURL.Download(ctx, opts.startOffset, opts.length, azblob.BlobAccessConditions{}, false, azblob.ClientProvidedKeyOptions{})
+	resp, err := az.clnt.DownloadStream(ctx, az.Bucket, az.getDest(object), &azblob.DownloadStreamOptions{
+		Range: blob.HTTPRange{Offset: opts.startOffset, Count: opts.length},
+	})
 	if err != nil {
-		return nil, azureToObjectError(err, az.Bucket, object)
+		return nil, azureToObjectError(err, az.Bucket, az.getDest(object))
 	}
 
-	rc := blob.Body(azblob.RetryReaderOptions{})
-	return rc, nil
+	return resp.Body, nil
 }
 
 func (az *warmBackendAzure) Remove(ctx context.Context, object string, rv remoteVersionID) error {
-	blob := az.serviceURL.NewContainerURL(az.Bucket).NewBlobURL(az.getDest(object))
-	_, err := blob.Delete(ctx, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
-	return azureToObjectError(err, az.Bucket, object)
+	_, err := az.clnt.DeleteBlob(ctx, az.Bucket, az.getDest(object), &azblob.DeleteBlobOptions{})
+	return azureToObjectError(err, az.Bucket, az.getDest(object))
 }
 
 func (az *warmBackendAzure) InUse(ctx context.Context) (bool, error) {
```

Comment:
```go
func (az *warmBackendAzure) InUse(ctx context.Context) (bool, error) {
        maxResults := int32(1)
        pager := az.clnt.NewListBlobsFlatPager(az.Bucket, &azblob.ListBlobsFlatOptions{
                Prefix:     &az.Prefix,
                MaxResults: &maxResults,
        })
        if !pager.More() {
                return false, nil
        }

        resp, err := pager.NextPage(ctx)
        if err != nil {
                if strings.Contains(err.Error(), "no more pages") {
                        return false, nil
                }
                return false, azureToObjectError(err, az.Bucket, az.Prefix)
        }

        return len(resp.Segment.BlobItems) > 0, nil
}
```

this Azure SDK API is weird. This is how this will work ^